sixtyfps_compiler/generator/cpp.rs

```diff
@@ -942,7 +944,17 @@ fn generate_component(
             Declaration::Function(Function {
                 name: "hide".into(),
                 signature: "() const".into(),
-                statements: Some(vec!["window.hide();".into()]),
+                statements: Some(vec!["m_window.hide();".into()]),
+                ..Default::default()
+            }),
+        ));
+
+        component_struct.members.push((
+            Access::Public,
+            Declaration::Function(Function {
+                name: "window".into(),
+                signature: "() const -> const sixtyfps::Window&".into(),
```

Comment:
This is a little awkward, in the sense that the window accessor is `const` and returns a `const &`, which means that functions like `set_position` would also have to be `const`.

I think a more idiomatic C++ API would be actually a `std::shared_ptr<sixtyfps::Window>`. So "clone" like before, but it's visible in the type name that it's explicitly shared and thus cloning won't create a new window on the screen. What do you think?

Reply:
how about returning a  `sixtyfps::Window&`  (not const)

Reply:
Or a  ` sixtyfps::Window*`

Reply:
Yes, returning a mutable reference is what I'm going to do now.